src/bindings/python/src/pyopenvino/graph/partial_shape.cpp

```diff
@@ -43,6 +43,26 @@ void regclass_graph_PartialShape(py::module m) {
                        :return: A PartialShape with the given rank (or undefined rank if not provided), and all dimensions are dynamic.
                     )");
 
+    auto check_shape_equality = [](const ov::PartialShape& self, const auto& container) {
+        if (self.rank().is_static() && self.rank().get_length() != container.size()) {
+            return false;
+        }
+        for (size_t i = 0; i < self.rank().get_length(); ++i) {
```

Comment:
I'm not certain this will work for a `PartialShape` with dynamic rank - accessing length of a dynamic dimension will probably cause an error.

Please check and add an appropriate test.